enclave/core/sgx/td.c

```diff
@@ -168,6 +168,32 @@ oe_sgx_td_t* oe_sgx_get_td()
     return td;
 }
 
+/*
+**==============================================================================
+**
+** oe_sgx_set_td_exception_handler_stack()
+**
+**     Internal API that allows an enclave to setup stack area for
+**     exception handlers to use.
+**
+**==============================================================================
+*/
+bool oe_sgx_set_td_exception_handler_stack(void* stack, uint64_t size)
+{
+    oe_sgx_td_t* td = oe_sgx_get_td();
+
+    /* ensure the stack + size is 16-byte aligned */
+    if (((uint64_t)stack + size) % 16)
+        return false;
+
+    td->exception_handler_stack_size = size;
+    /* Put a barrier to prevent the compiler reorders the operations */
```

Comment:
```suggestion
    /* Put a barrier to prevent the compiler reordering of the assignment */
```

Reply:
Also document why the order is important. In the comment cross refer to enter.S

Reply:
now we check size != 0, no longer needs this